test/module/irohad/common/raw_block_loader_test.cpp

```diff
@@ -15,32 +15,40 @@
  * limitations under the License.
  */
 
-#include "main/raw_block_loader.hpp"
 #include <gtest/gtest.h>
 
-#include "framework/test_block_generator.hpp"
-#include "model/converters/json_block_factory.hpp"
-#include "model/converters/json_common.hpp"
-#include "model/converters/pb_command_factory.hpp"
+#include "interfaces/iroha_internal/block.hpp"
+#include "main/raw_block_loader.hpp"
 
-using namespace iroha::main;
-using namespace iroha::model::converters;
-using namespace iroha;
+using iroha::main::BlockLoader;
 
 /**
- * @given generated block
+ * @given json of a block
  *
- * @when convert block to JSON
- * AND parseBlock() with BlockLoader API
+ * @when convert json to block using raw block loader
  *
- * @then check that blocks are equal
+ * @then check that block is correct
  */
 TEST(BlockLoaderTest, BlockLoaderWhenParseBlock) {
   BlockLoader loader;
-  auto block = framework::generator::generateBlock();
-  auto doc = JsonBlockFactory().serialize(block);
-  auto str = jsonToString(doc);
-  auto new_block = loader.parseBlock(str);
-  ASSERT_TRUE(new_block);
-  ASSERT_EQ(block, *new_block);
+  auto str =
+      R"({
+"payload": {
+  "transactions": [],
+  "height": 1,
+  "prev_block_hash": "000000000000000000000000000000000000000000000000000000000000000",
```

Comment:
Hash hasn't been checked

Reply:
There is an issue with a hash. It is stored inside a block as crypto::Hash, and its representation is different from the passed string. That is why we cannot compare hash inside a block with the one in json.

Reply:
`crypto::Hash(std::string(32, '\x00'))`?